pyadjoint/drivers.py

```diff
@@ -18,8 +18,8 @@ def compute_gradient(J, m, options=None, tape=None, adj_value=1.0):
         OverloadedType: The derivative with respect to the control. Should be an instance of the same type as
             the control.
     """
-    options = {} if options is None else options
-    tape = get_working_tape() if tape is None else tape
+    options = options or {}
```

Comment:
I agree that this is cleaner - generally though, I'd be wary about "cleaning" code in places you aren't changing. It's likely to introduce new bugs.